tests/QueryBuilderTest.php

```diff
@@ -209,4 +260,24 @@ public function it_queries_the_correct_data_for_a_relationship_query()
         $this->assertEquals(789, $queryBuilderResult->id);
         $this->assertEquals(123, $queryBuilderResult->testModel->id);
     }
+
+    /** @test */
+    public function it_does_not_lose_pivot_values_with_belongs_to_many_relation()
+    {
+        /** @var TestModel $testModel */
+        $testModel = TestModel::create(['id' => 324, 'name' => 'Illia']);
+
+        /** @var RelatedThroughPivotModel $relatedThroughPivotModel */
+        $relatedThroughPivotModel = RelatedThroughPivotModel::create(['id' => 721, 'name' => 'Kate']);
+
+        $testModel->relatedThroughPivotModelsWithPivot()->attach($relatedThroughPivotModel, ['location' => 'Wood Cottage']);
+
+        $foundTestModel = QueryBuilder::for($testModel->relatedThroughPivotModelsWithPivot())
+            ->first();
+
+        $this->assertSame(
+            'Wood Cottage',
+            $foundTestModel->pivot->location
+        );
+    }
```

Comment:
Here we ensure that the PR was not submitted in vain (pivot values are not lost).